iphone/TitaniumKit/TitaniumKit/Sources/API/Webcolor.m

```diff
@@ -30,11 +32,45 @@ + (UIColor *)checkmarkColor
   return checkmarkColor;
 }
 
+
++ (UIColor *)semanticColorNamed:(NSString *)colorName
+{
+  if ([TiUtils isIOSVersionOrGreater:@"11.0"]) {
+    return [UIColor colorNamed:colorName];
+  }
+
+  if (semanticColors == nil) {
+    NSString *colorsPath = [NSBundle.mainBundle pathForResource:@"semantic.colors" ofType:@"json"];
+    if (![NSFileManager.defaultManager fileExistsAtPath:colorsPath]) {
+      return nil;
+    }
+
+    NSData *colors = [NSData dataWithContentsOfFile:colorsPath options:NSDataReadingMappedIfSafe error:nil];
+    semanticColors = [NSJSONSerialization JSONObjectWithData:colors options:NSJSONReadingMutableContainers error:nil];
+  }
+
+  NSDictionary *colorMap = semanticColors[colorName];
+  if (colorMap == nil) {
+    return nil;
+  }
+
+  NSString *currentTraitCollection = TiApp.controller.traitCollection.userInterfaceStyle == UIUserInterfaceStyleDark ? @"dark" : @"light";
```

Comment:
This new chunk of code feels like it should live in the `common` JS area, but I don't see how we could do that and have it effectively hijack all the possible places where colors are set (i.e. all the proxies with various color properties).

Reply:
Yep, that was the reason to do it here. And since the JSON only needs to be loaded once, it doesn't cause an overhead to process the array (which wouldn't be a problem on the JS side as well though).